### ISSUE TITLE
Requesting a minimal working example for parallel Wikies

### ISSUE THREAD

**Issue by llinfeng** [vimwiki/vimwiki#365]
## Request
While there has been a good assortment of shortcuts and functions implemented in Vimwiki that help with managing multiple Wikis, it bugged me quite a bit to achieve the following:
* Keep two Wikis in parallel, and
  * I can use `<leader>w<leader>w` to open diary for main Wiki, and `2<leader>w<leader>w` for another Wiki.
  * The two Wiki-directories are not necessarily nested, but Wiki-entries in the two directories shall at best be mutually accessible (through shortcuts like `[[file:../Other_Wiki/filename#Section1]]`)

## Sample setting (not yet working)
I have my wiki-settings implemented from years ago, where I was having in mind three separate Wikis. I don't quite understand most of the settings, but I would like to keep the syntax (each setting in a line of its own).

````
let wiki_1 = {}
let wiki_1.index = 'Academia'
" This option is turned down since Vimwiki cannot convert markdown syntax to html.
"let wiki_1.syntax = 'markdown'
let wiki_1.path = '~/Nutstore/Wiki/Warehouse/'
let wiki_1.diary_rel_path = ''
let wiki_1.diary_index = 'Journal'
let wiki_1.path_html = '~/Nutstore/Wiki/html'
let wiki_1.html_header = '~/Nutstore/Wiki/html'
let wiki_1.template_path = '~/Dropbox/Shu/HTML/Template-Defualt'
let wiki_1.template_default = 'default'
let wiki_1.template_ext = '.html'
let wiki_1.auto_export = 0
" Second wiki, for personal use.
let wiki_2 = {}
let wiki_2.index = 'Life'
let wiki_2.path = '~/Nutstore/Wiki/Warehouse/Diary'
let wiki_2.diary_rel_path = ''
let wiki_2.diary_index = 'Journal'
let wiki_2.path_html = '~/Nutstore/Wiki/html'
let wiki_2.html_header = '~/Nutstore/Wiki/html'
let wiki_2.template_path = '~/Dropbox/Shu/HTML/Template-Defualt'
let wiki_2.template_default = 'default'
let wiki_2.template_ext = '.html'
let wiki_2.auto_export = 0
" Third wiki, for working purpose (not heavily used.)
let wiki_3 = {}
let wiki_3.index = 'Work'
let wiki_3.path = '~/Nutstore/Wiki/Warehouse/'
let wiki_3.path_html = '~/Nutstore/Wiki/html'
let wiki_3.html_header = '~/Nutstore/Wiki/html'
let wiki_3.template_path = '~/Dropbox/Shu/HTML/Template-Defualt'
let wiki_3.template_default = 'default'
let wiki_3.template_ext = '.html'
let wiki_3.auto_export = 0
" This is the "make" commend that constructs the two wiki's.
let wiki_1.nested_syntaxes = {'python': 'python', 'c++': 'cpp', 'vim': 'vim', 'R': 'r', 'bash': 'sh', 'ahk': 'autohotkey'}
let wiki_2.nested_syntaxes = {'python': 'python', 'c++': 'cpp', 'vim': 'vim', 'R': 'r', 'bash': 'sh', 'ahk': 'autohotkey'}
let wiki_3.nested_syntaxes = {'python': 'python', 'c++': 'cpp', 'vim': 'vim', 'R': 'r', 'bash': 'sh', 'ahk': 'autohotkey'}
" Construct the wiki
let g:vimwiki_list = [wiki_1, wiki_2, wiki_3]
````

### Problem with the sample setting
1. `2<leader>w<leader>w` does literally nothing when pressed for the first time; to get to the desired page/file/location/wiki, I need to pressed a second time. This equals to 10 keystrokes from which half could have been saved.
2. Similarly, `2<Leader>wi` does not create the `Jounral.wiki` in the specified directory "`'~/Nutstore/Wiki/Warehouse/Diary'`". Instead, it sends me to the master Wiki (the first wiki).

### Question
Should I avoid using a nested folder for the second Wiki? A minimal working example with both shortcuts working would be great!

**Comment by malramsay64** [comment]
Have a look at #366. I was having similar issues and have gone and fixed it. Well it works for me, it will hopefully solve the problems you are having.

Also while I am here, to make your wiki settings simpler you can set a default set of settings, which is then copied and edited for each wiki. See my wiki config below for an example

```
" Wiki settings
let wiki_default = {}
let wiki_default.auto_export = 0
let wiki_default.auto_toc = 0
let wiki_default.nested_syntaxes = {'python': 'python', 'c++': 'cpp', 'sh': 'sh'}

let home_wiki = copy(wiki_default)
let home_wiki.path = '~/Documents/notes/personal/'
let home_wiki.diary_rel_path = 'diary/'
let home_wiki.path_html = '~/Documents/notes/html/personal/'

let phd_wiki = copy(wiki_default)
let phd_wiki.path = '~/Documents/notes/PhD/'
let phd_wiki.diary_rel_path = 'logbook/'

let g:vimwiki_list = [home_wiki, phd_wiki]
```

**Comment by ipstone** [comment]
Is there a way to use vimwiki based on folder? Instead of base on the configurations? - such as looking for the default files/setup/path in a folder?

**Comment by ranebrown** [comment]
> Is there a way to use vimwiki based on folder? Instead of base on the configurations? - such as looking for the default files/setup/path in a folder?

Not currently, feel free to open a new issue.
